### PR TITLE
Science Arm Code Integration

### PR DESCRIPTION
**Pull request by bli76**
- Split MRoverArm class into StandardArm and ScienceArm
- Updated build system to choose which arm to use
- Change Vector6d to vector<double>
- Change floating number 6 to joints.size()

### CODE REVIEW

File: jetson/ra_kinematics/README.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+Subscriber: jetson/nucleo_bridge
+
+#### FK Transform \[Publisher\] "/fk_transform" ####
+Message: [FKTransform.lcm](https://github.com/umrover/mrover-workspace/blob/master/rover_msgs/FKTransform.lcm) \
+Publisher: jetson/ra_kinematics \
+Subscriber: base_station/kineval_stencil
+
+### LCM Subscriptions ###
+
+#### Arm Position \[Subscriber\] "/arm_position" ####
```
Comment (tabiosg) [inline jetson/ra_kinematics/README.md:80]: should be ra_position

File: jetson/ra_kinematics/README.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+Subscriber: jetson/ra_kinematics
+
+When in Simulation Mode, jetson/ra_kinematics publishes to /arm_position to simulate receiving arm position message from nucleo_bridge.
+
+#### Debug Message \[Publisher\] "/debug_message" ####
+Message: [DebugMessage.lcm](https://github.com/umrover/mrover-workspace/blob/master/rover_msgs/DebugMessage.lcm) \
+Publisher: jetson/ra_kinematics \
+Subscriber: base_station/kineval_stencil
+
+#### RA Closed Loop \[Publisher\] "/ik_ra_control" ####
```
Comment (tabiosg) [inline jetson/ra_kinematics/README.md:68]: should be ra_ik_cmd

File: jetson/ra_kinematics/README.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+### Testing ###
+
+The ra_kinematics package uses the EECS 280 testing framework. Testing files are found in the test directory.
+
+To test the package, run `$ make tests ` from the ra_kinematics directory. To test more specific files, see the Makefile for more commands.
+
+### LCM Publications ###
+
+#### Arm Position \[Publisher\] "/arm_position" ####
```
Comment (tabiosg) [inline jetson/ra_kinematics/README.md:56]: should be ra_position

File: jetson/ra_kinematics/main.cpp [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+    //     robot_arm = ScienceArm(geom, lcmObject);
+    // }
+    
+    // (science) ? ScienceArm(geom, lcmObject) : StandardArm(geom, lcmObject);
+    if (science) {
+        ScienceArm robot_arm(geom, lcmObject);
+
+        lcmHandlers handler(&robot_arm);
+
+        lcmObject.subscribe( "/arm_position", &lcmHandlers::armPositionCallback, &handler );
```
Comment (tabiosg) [inline jetson/ra_kinematics/main.cpp:118]: should be ra_position

File: jetson/ra_kinematics/main.cpp [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+        send_arm_position.join();
+
+        return 0;
+    }
+    else {
+        StandardArm robot_arm(geom, lcmObject);
+
+        lcmHandlers handler(&robot_arm);
+
+        lcmObject.subscribe( "/arm_position", &lcmHandlers::armPositionCallback, &handler );
```
Comment (tabiosg) [inline jetson/ra_kinematics/main.cpp:144]: ra_position

File: jetson/ra_kinematics/main.cpp [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+
+        execute_spline.join();
+        send_arm_position.join();
+
+        return 0;
+    }
+
+    // lcmHandlers handler(&robot_arm);
+
+    // lcmObject.subscribe( "/arm_position", &lcmHandlers::armPositionCallback, &handler );
```
Comment (tabiosg) [inline jetson/ra_kinematics/main.cpp:168]: ra_position

File: jetson/ra_kinematics/mrover_arm.cpp [hunk 7]
```diff
@@ -0,0 +1,747 @@
...
+        arm_position.joint_b = config[1];
+        arm_position.joint_c = config[2];
+        arm_position.joint_d = config[3];
+        arm_position.joint_e = config[4];
+        arm_position.joint_f = config[5];
+    }
+    else {
+        arm_position.joint_a = config[0];
+        arm_position.joint_b = config[1];
+        arm_position.joint_c = config[2];
```
Comment (tabiosg) [inline jetson/ra_kinematics/mrover_arm.cpp:563]: When does it enter this else statement? Is it for the science arm? If so, then the members should be double joint_a, b, c, and e. (e not d)

**Comment by bli76** [comment]
New PR was created with merge conflicts resolved on branch skorman-science-arm-2

**Pull request closed without merging**